util-core/src/main/scala/com/twitter/concurrent/AsyncStream.scala

```diff
@@ -279,13 +270,15 @@ final class AsyncStream[+A] private (private val underlying: Future[LazySeq[A]])
    * never.scanLeft(z)(f) == z +:: never // logical equality
    * }}}
    */
-  def scanLeft[B](z: B)(f: (B, A) => B): AsyncStream[B] = z +:: AsyncStream(
-    underlying.flatMap {
-      case Empty => empty[B].underlying
-      case One(a) => of(f(z, a)).underlying
-      case cons: Cons[A] => cons.tail.scanLeft(f(z, cons.head))(f).underlying
+  def scanLeft[B](z: B)(f: (B, A) => B): AsyncStream[B] =
+    this match {
+      case Embed(fas) => Embed(fas.map(_.scanLeft(z)(f)))
```

Comment:
Ah, I put embed first because that's the only time we don't want to prefix with z. What do you think if I added a comment about that and left it in that first position? Open to other formatting ideas too.